crates/context_aware_config/src/api/context/helpers.rs

```diff
@@ -57,14 +57,43 @@ pub fn validate_condition_with_mandatory_dimensions(
     Ok(())
 }
 
+pub fn validate_condition_with_dependent_dimensions(
+    conn: &mut DBConnection,
+    context_map: &Map<String, Value>,
+    schema_name: &SchemaName,
+) -> superposition::Result<()> {
+    use dimensions::dsl;
+    let keys = context_map.keys();
+    let dependency_lists: Vec<DependencyGraph> = dsl::dimensions
+        .filter(dsl::dimension.eq_any(keys))
+        .select(dsl::dependency_graph)
+        .schema_name(schema_name)
+        .load::<DependencyGraph>(conn)?;
+    let dependency_list: HashSet<String> = dependency_lists
+        .iter()
+        .flat_map(|obj| obj.keys())
+        .cloned()
+        .collect();
```

Comment:
these lines feel like to be a good candidate for a utility function, something like:
```suggestion
    fn get_all_dependants(dimensions: Vec<String>) -> superposition::Result<HashSet<String>> {
        let dependency_lists: Vec<DependencyGraph> = dsl::dimensions
        .filter(dsl::dimension.eq_any(dimensions))
        .select(dsl::dependency_graph)
        .schema_name(schema_name)
        .load::<DependencyGraph>(conn)?;
    dependency_lists
        .iter()
        .flat_map(|obj| obj.keys())
        .cloned()
        .collect()
   }
```